### PR TITLE
perf: Only render the TimezoneChangeDialog when needed

### PR DESCRIPTION
**Pull request by emrysal**
## What does this PR do?

* Improves various interfaces in this file.
* This was rendered on every page in the web app. - unnecessary overhead.

Before:
image

After:
image

### CODE REVIEW

File: packages/features/settings/TimezoneChangeDialog.tsx [hunk 1]
```diff
@@ -4,109 +4,122 @@ import { useSession } from "next-auth/react";
...
+      {/* Dismiss dialog for a day on outside interaction */}
+      <DialogContent type="creation" onInteractOutside={handleInteractOutside}>
+        <TimezoneChangeDialogContent />
+      </DialogContent>
     </Dialog>
   );
 }
+
+export default function TimezoneChangeDialog() {
+  const [renderDialog, setRenderDialog] = useState(false);
```
Comment (emrysal) [inline packages/features/settings/TimezoneChangeDialog.tsx:109]: Wait for window to be ready (and the cookie to be available) before even starting to consider doing timezone checking.

File: packages/features/settings/TimezoneChangeDialog.tsx [hunk 2]
```diff
@@ -4,109 +4,122 @@ import { useSession } from "next-auth/react";
...
+
+  const handleInteractOutside = () => {
+    hideDialogFor([1, "day"], t("we_wont_show_again"));
+  };
+
   return (
     <Dialog open={open} onOpenChange={setOpen}>
-      <TimezoneChangeDialogContent browserTimezone={browserTimezone} onAction={() => setOpen(false)} />
+      {/* Dismiss dialog for a day on outside interaction */}
+      <DialogContent type="creation" onInteractOutside={handleInteractOutside}>
```
Comment (emrysal) [inline packages/features/settings/TimezoneChangeDialog.tsx:101]: Use DialogContent separately from TimezoneChangeDialogContent. Using this type of wrapper we ensure that TimezoneChangeDialogContent is only rendered when the DialogContent isOpen. Before the underlying component was always rendered.

File: packages/features/settings/TimezoneChangeDialog.tsx [hunk 3]
```diff
@@ -4,109 +4,122 @@ import { useSession } from "next-auth/react";
...
+  const { open, setOpen, browserTimezone } = useOpenTimezoneDialog();
+
+  if (!isLocaleReady) return null;
+
   const utils = trpc.useUtils();
   const formattedCurrentTz = browserTimezone.replace("_", " ");
 
-  // save cookie to not show again
-  function onCancel(hideFor: [number, dayjs.ManipulateType], toast: boolean) {
-    onAction("cancel");
```
Comment (emrysal) [inline packages/features/settings/TimezoneChangeDialog.tsx:24]: Remove very dirty interface, also with onAction("update") - these onActions which used to dismiss the Dialog are now handled automatically by event bubbling from DialogClose.

**Review by sean-brydon — APPROVED** [review]
LGTM nice fix!

**Pull request merged**